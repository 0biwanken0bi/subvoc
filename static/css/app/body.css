html,
body {
    height: 100%;
    background: -webkit-linear-gradient(125deg, #7b4397, #dc2430);
    background: linear-gradient(125deg, #7b4397, #dc2430);
    background-attachment: fixed;
}

body {
    color: #fff;
    text-align: center;
    text-shadow: 0 .05rem .1rem rgba(0, 0, 0, .5);
}

body .wrapper {
    display: block;
    overflow: hidden;
    position: relative;
    margin: 0 auto;
    width: 100%;
}

@media (min-width: 40rem) {
    .container {
        max-width: 60rem;
    }
}
