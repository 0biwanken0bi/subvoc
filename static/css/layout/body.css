html,
body {
    height: 100%;
    background: -webkit-linear-gradient(right, #7b4397, #dc2430);
    background: linear-gradient(to left, #7b4397, #dc2430);
}

body {
    color: #fff;
    text-align: center;
    text-shadow: 0 .05rem .1rem rgba(0, 0, 0, .5);
}

body .wrapper {
    display: block;
    overflow: hidden;
    position: relative;
    margin: 0 auto;
    width: 100%;
}

.wrapper>.container {
    padding-top: 7.5rem;
}

@media (min-width: 80rem) {
  body .wrapper {
      width: 80rem;
  }
}

