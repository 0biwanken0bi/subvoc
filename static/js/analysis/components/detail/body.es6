import $ from 'jquery';
import preact from 'preact';

import { WordDefinitionList } from './definition.es6';
import { WordExcerptList } from './excerpt.es6';
import { WordPartOfSpeachHeader } from './selector.es6';


class WordDetailBody extends preact.Component {
    render({ lookup, selection, onSelectPOS }) {
        const headers = [['noun', 'noun'],
                         ['verb', 'verb'],
                         ['adjective', 'adj'],
                         ['adverb', 'adv']];

        const selectedPOS = selection.POS || $.grep(headers, (h) => lookup[h[0]])[0][0];
        const wordForPOS = selection.word.byPOS[selectedPOS];

        console.log(selection.word);

        return <div>
            <header class="tab-group">
                { $.map(headers, (header) =>
                    <WordPartOfSpeachHeader
                        active={selectedPOS === header[0]}
                        enabled={lookup[header[0]]}
                        code={header[0]}
                        label={header[1]}
                        freq={(selection.word.byPOS[header[0]] || {}).freq}
                        onSelectPOS={onSelectPOS} />
                ) }
            </header>

            <section>
                <WordExcerptList excerpts={(wordForPOS || {}).excerpts} />
            </section>

            <section class="explanations">
                <WordDefinitionList definitions={lookup[selectedPOS]} />
            </section>
        </div>;
    }
}


export { WordDetailBody };
