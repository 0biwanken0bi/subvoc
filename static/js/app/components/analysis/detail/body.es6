import find from 'array.prototype.find';
import React from 'react';

import { WordDefinitionList } from './definition.es6';
import { WordExcerptList } from './excerpt.es6';
import { WordPartOfSpeachHeader } from './pos.es6';



const PARTS_OF_SPEACH = ['noun', 'verb', 'adj', 'adv'];


function getExcerpts(word, pos) {
    return (word.byPOS[pos] || {}).excerpts || [];
}

function getFreq(word, pos) {
    return (word.byPOS[pos] || {}).freq || 0;
}

function getDefinitions(word, pos) {
    return word.lookup[pos] || [];
}

function hasDetails(word, pos) {
    return getDefinitions(word, pos).length > 0 || getExcerpts(word, pos).length > 0;
}


const WordDetailBody = ({ selection, onSelectPOS }) => {
    const selectedPOS = selection.POS ||
        find(PARTS_OF_SPEACH, (pos) => getExcerpts(selection.word, pos).length > 0)[0];

    return <div>
        <header className="tab-group">
            { PARTS_OF_SPEACH.map((pos) => {
                return <WordPartOfSpeachHeader
                    key={pos}
                    active={selectedPOS === pos}
                    enabled={hasDetails(selection.word, pos)}
                    label={pos}
                    freq={getFreq(selection.word, pos)}
                    onSelectPOS={onSelectPOS} />;
            } ) }
        </header>

        <section>
            <WordExcerptList excerpts={getExcerpts(selection.word, selectedPOS)} />
        </section>

        <section>
            <WordDefinitionList definitions={getDefinitions(selection.word, selectedPOS)} />
        </section>
    </div>;
};


export { WordDetailBody };
