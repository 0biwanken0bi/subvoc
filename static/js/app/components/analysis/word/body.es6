import find from 'array.prototype.find';
import React from 'react';

import { WordDefinitionList } from './definition.es6';
import { WordExcerptList } from './excerpt.es6';
import { getDefinitions, getExcerpts, PARTS_OF_SPEACH, PartOfSpeachSelector } from './selector.es6';


class WordDetailBody extends React.Component {

    constructor(props) {
        super(props);
        this.state = {};
    }

    handleSelectPOS(POS) {
        this.setState((prevState) => {  
            prevState.POS = POS;
        });
    }

    getSelectedPOS(word) {
        return this.state.POS ||
            find(PARTS_OF_SPEACH, (pos) => getExcerpts(word, pos).length > 0);
    }


    render() { 
        const { word } = this.props;
        const selectedPOS = this.getSelectedPOS(word);

        return <div>
            <header className="tab-group">
                <PartOfSpeachSelector 
                    word={word} 
                    selected={selectedPOS} 
                    onSelect={(p) => this.handleSelectPOS(p)} />
            </header>

            <section>
                <WordExcerptList excerpts={getExcerpts(word, selectedPOS)} />
            </section>

            <section>
                <WordDefinitionList definitions={getDefinitions(word, selectedPOS)} />
            </section>
        </div>;
    }
}


export { WordDetailBody };
