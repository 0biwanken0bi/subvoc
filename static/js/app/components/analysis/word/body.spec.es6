import React from 'react';

import { shallow } from 'enzyme';

import { WordDetailBody } from './body.es6';


describe("<WordDetailBody />", () => {
    
    test("should render with pre-selected POS", () => {
        const word = {
            byPOS: {
                noun: { freq: 1, excerpts: ['...'] }
            },
            lookup: {}
        };

        const result = shallow(
            <WordDetailBody POS='noun' word={word} />
        );

        expect(result).toMatchSnapshot();
    });

    test("should render by picking POS with excerpt", () => {
        const word = {
            byPOS: {
                verb: { freq: 1, excerpts: ['...'] }
            },
            lookup: {
                noun: ['...']
            }
        };

        const result = shallow(
            <WordDetailBody word={word} />
        );

        expect(result).toMatchSnapshot();
    });
});