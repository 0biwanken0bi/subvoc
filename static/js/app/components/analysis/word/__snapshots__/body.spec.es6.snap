// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordDetailBody /> should render by picking POS with excerpt 1`] = `
<div>
  <header
    className="tab-group"
  >
    <PartOfSpeachSelector
      onSelect={[Function]}
      selected="verb"
      word={
        Object {
          "byPOS": Object {
            "verb": Object {
              "excerpts": Array [
                "...",
              ],
              "freq": 1,
            },
          },
          "lookup": Object {
            "noun": Array [
              "...",
            ],
          },
        }
      }
    />
  </header>
  <section>
    <WordExcerptList
      excerpts={
        Array [
          "...",
        ]
      }
    />
  </section>
  <section>
    <WordDefinitionList
      definitions={Array []}
    />
  </section>
</div>
`;

exports[`<WordDetailBody /> should render with pre-selected POS 1`] = `
<div>
  <header
    className="tab-group"
  >
    <PartOfSpeachSelector
      onSelect={[Function]}
      selected="noun"
      word={
        Object {
          "byPOS": Object {
            "noun": Object {
              "excerpts": Array [
                "...",
              ],
              "freq": 1,
            },
          },
          "lookup": Object {},
        }
      }
    />
  </header>
  <section>
    <WordExcerptList
      excerpts={
        Array [
          "...",
        ]
      }
    />
  </section>
  <section>
    <WordDefinitionList
      definitions={Array []}
    />
  </section>
</div>
`;
